From 058f2d23321839e4aca1af7eb4537883fe7dd92a Mon Sep 17 00:00:00 2001
From: Max Kellermann <max.kellermann@gmail.com>
Date: Tue, 17 Oct 2017 15:56:06 +0200
Subject: [PATCH 2/2] mpdclient: check source again after mpd_glib_leave()

Just in case mpd_glib_leave() has found an error and had to close the
connection, check it again to avoid an assertion failure (or crash) in
mpdclient_schedule_enter_idle().

Closes #9
---
 NEWS            | 1 +
 src/mpdclient.c | 3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/mpdclient.c b/src/mpdclient.c
index 6a81a0b..1801ad2 100644
--- a/src/mpdclient.c
+++ b/src/mpdclient.c
@@ -490,7 +490,8 @@ mpdclient_get_connection(struct mpdclient *c)
 		c->idle = false;
 		mpd_glib_leave(c->source);
 
-		mpdclient_schedule_enter_idle(c);
+		if (c->source != NULL)
+			mpdclient_schedule_enter_idle(c);
 	}
 
 	return c->connection;
-- 
2.14.2

